scripts/java-api-check-all.sh

```diff
@@ -38,17 +38,17 @@ CURRENT_BRANCH=$(git rev-parse --abbrev-ref HEAD)
 TARGET_BRANCH=$(gh pr view "$CURRENT_BRANCH" --json baseRefName --jq '.baseRefName')
 echo "CURRENT_BRANCH: $CURRENT_BRANCH"
 echo "TARGET_BRANCH: $TARGET_BRANCH"
-readonly RELEASE_TAG_PATTERN="v10\.[0-9]*\.[0-9]*$"
 readonly STABLE_RELEASE_TAG_PATTERN="v10\.[0-9]*\.0$"
 
 if ! [ "$TARGET_BRANCH" = "main" ]; then # Release branches
-  LATEST_RELEASE_TAG_IN_THIS_BRANCH=$(git tag --merged "$CURRENT_BRANCH" --sort=-creatordate | grep $RELEASE_TAG_PATTERN | head -n 1)
-  if [ "$LATEST_RELEASE_TAG_IN_THIS_BRANCH" = "*beta*" ] || [ "$LATEST_RELEASE_TAG_IN_THIS_BRANCH" = "*rc*" ]; then
-    # Use the latest stable minor release tag during rc and beta release
+  # filter tags present in the current release branch
+  LATEST_RELEASE_TAG_IN_THIS_BRANCH=$(git tag --sort=-creatordate | { grep "^$TARGET_BRANCH" || test $? = 1; } | head -n 1) # grep sets error if result is empty, so test is needed
```

Comment:
Last tag in the release branch, named with the release branch prefix is used to run comparison. 
e.g. for the current branch `v10.9` we take the last created `v10.9.0-rc.1`. 
If there's no tag (e.g. first beta release is created) - previous stable release will be used for comparison.

Reply:
shouldn't we always compare to the last stable minor release?

Reply:
That's an open question. My thinking here was - if `10.9.0-beta.1` is compatible with the previous `10.8.5` - then `10.9.0-rc.1` will be as well. Then it should be enough to compare `10.9.0-rc.1` and `10.9.0-beta.1`. WDYT?

Reply:
Reworked to correctly find the previous minor release